spec/jekyll_commonmark_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe(Jekyll::Converters::Markdown::CommonMark) do
+  let(:options) { [] }
+  let(:extensions) { [] }
+  let(:config) do
+    {
+      "commonmark" => {
+        "options"    => options,
+        "extensions" => extensions,
+      },
+    }
+  end
+  let(:commonmark) { described_class.new(config) }
+  let(:output) { commonmark.convert(content) }
+
+  context "with default configuration" do
+    it "produces the correct script tag" do
+      actual = commonmark.convert("# Heading")
+      expected = "<h1>Heading</h1>"
+      expect(actual).to match(expected)
+    end
+
+    it "does not treat newlines as hardbreaks" do
+      actual = commonmark.convert("a\nb")
+      expected = "<p>a\nb</p>"
+      expect(actual).to match(expected)
+    end
+
+    it "treats double linebreaks as a new paragraph" do
+      actual = commonmark.convert("a\n\nb")
+      expected = "<p>a</p>\n<p>b</p>"
+      expect(actual).to match(expected)
+    end
+
+    it "escapes quotes" do
+      actual = commonmark.convert('"SmartyPants"')
+      expected = "<p>&quot;SmartyPants&quot;</p>"
+      expect(actual).to match(expected)
+    end
+
+    it "does not link urls" do
+      actual = commonmark.convert("https://example.com")
+      expected = "https://example.com"
+      expect(actual).to match(expected)
+    end
```

Comment:
IMO, there should also be a test for Markdown links
```ruby
commonmark.convert("[example](https://example.com)")
commonmark.convert("[example][link]\n\n[link]: https://example.com")
```

Reply:
Remember that we're not testing the MarkDown processor itself, merely our _interface_ to the MarkDown processor. If the MarkDown processor does not conform to the CommonMark spec, its own tests will reveal that. :+1:

That said, if there is a reason to add this test, it would not be difficult.

Reply:
Ah I see.. is there any aspect where our *interface* **markedly differs** from `kramdown`? If yes, then that should definitely be tested (will serve as documentation) because the HTML markup is going to change when users switch to this parser..